lib/language_pack/ruby.rb

```diff
@@ -219,34 +219,46 @@ def ruby_version
   # default JAVA_OPTS
   # return [String] string of JAVA_OPTS
   def default_java_opts
-    "-Xss512k -XX:+UseCompressedOops -Dfile.encoding=UTF-8"
+    '-Dfile.encoding=UTF-8'
+  end
+
+  def set_jvm_stack_size
+    ENV['JVM_STACK_SIZE'] = env('JVM_STACK_SIZE') || 512
```

Comment:
Here's the bug. This method `set_jvm_stack_size` returns back a string of "512"

Then later this is called here:

```
        ENV["JAVA_STACK"] = run(<<-SHELL).chomp
#{set_jvm_stack_size}
echo #{default_java_stack_mem}
```

it ends up being a raw `512` on that line which bash interprets as a variable, one that does not exist. It causes this `JAVA_STACK` to be set to:

```
bash: 512: command not found
-Xss512k
```

In my debugging branch, i changed those `run` invocations to `run!` so they will error early, but it didn't trigger. Is that because there's no `set -e` at the  top of this script @jkutner ?

@partounian what do you want to do to fix this up? Is the goal simply to set that environment variable? If so, we could run it outside of the shell script, alternatively we could return a nil?

Reply:
Also FWIW you should be able to run this test locally and see it fail:

```
$ rspec ./spec/hatchet/rubies_spec.rb:72
```

Reply:
I think the return of `set_jvm_stack_size` needs to be a valid bash script as it's later added to profile.d

```
add_to_profiled set_jvm_stack_size
```